bee-storage/src/rocksdb/mod.rs

```diff
@@ -0,0 +1,541 @@
+
+        for (index, hash) in transaction_hashes.iter().enumerate() {
+            let res = db.get_cf(
+                &transaction_cf_hash_to_snapshot_index,
+                cast_slice(hash.to_inner().encode::<T5B1Buf>().as_i8_slice()),
+            )?;
+            if res.is_some() {
+                //We assume the absence of a value means the transaction is not known to be confirmed
+                let transaction_snapshot_index_buffer = res.unwrap();
+                unsafe { ptr::copy(transaction_snapshot_index_buffer.as_ptr(), u32_buffer.as_mut_ptr(), 4) };
+                solid_states[index] = u32::from_le_bytes(u32_buffer);
+            }
+        }
+
+        Ok(solid_states)
+    }
+
+    async fn delete_transactions(
+        &self,
+        transaction_hashes: &HashSet<bee_bundle::Hash>,
+    ) -> Result<(), RocksDbBackendError> {
+        let db = self.0.connection.db.as_ref().unwrap();
+        let mut batch = rocksdb::WriteBatch::default();
+        let transaction_cf_hash_to_transaction = db.cf_handle(TRANSACTION_CF_HASH_TO_TRANSACTION).unwrap();
+
+        for hash in transaction_hashes {
+            let hash_buf = hash.to_inner().encode::<T5B1Buf>();
+            batch.delete_cf(&transaction_cf_hash_to_transaction, cast_slice(hash_buf.as_i8_slice()))?;
+        }
+
+        let mut write_options = WriteOptions::default();
+        write_options.set_sync(false);
+        write_options.disable_wal(true);
+
+        db.write_opt(batch, &write_options)?;
+        Ok(())
+    }
+
+    async fn insert_milestone(&self, milestone: Milestone) -> Result<(), RocksDbBackendError> {
+        let db = self.0.connection.db.as_ref().unwrap();
+
+        let milestone_cf_hash_to_index = db.cf_handle(MILESTONE_CF_HASH_TO_INDEX).unwrap();
+        let milestone_cf_index_to_hash = db.cf_handle(MILESTONE_CF_INDEX_TO_HASH).unwrap();
+
+        let hash_buf = milestone.hash().to_inner().encode::<T5B1Buf>();
+        db.put_cf(
+            &milestone_cf_hash_to_index,
+            cast_slice(hash_buf.as_i8_slice()),
+            milestone.index().to_le_bytes(),
+        )?;
+
+        db.put_cf(
+            &milestone_cf_index_to_hash,
+            milestone.index().to_le_bytes(),
+            cast_slice(hash_buf.as_i8_slice()),
+        )?;
+        Ok(())
+    }
+
+    async fn find_milestone(&self, milestone_hash: bee_bundle::Hash) -> Result<Milestone, RocksDbBackendError> {
+        let db = self.0.connection.db.as_ref().unwrap();
+        let milestone_cf_hash_to_index = db.cf_handle(MILESTONE_CF_HASH_TO_INDEX).unwrap();
+        let res = db.get_cf(
+            &milestone_cf_hash_to_index,
+            cast_slice(milestone_hash.to_inner().encode::<T5B1Buf>().as_i8_slice()),
+        )?;
+
+        if res.is_none() {
+            return Err(RocksDbBackendError::TransactionDoesNotExist);
+        }
+
+        let mut index_buf: [u8; 4] = [0; 4];
+        unsafe { ptr::copy(res.unwrap().as_slice().as_ptr(), index_buf.as_mut_ptr(), 4) };
```

Comment:
Probably safer to use `copy_from_slice` instead
```suggestion
        index_buf.copy_from_slice(res.unwrap().as_slice());
```